core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -2766,10 +2792,15 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
 #if WASM_ENABLE_GC == 0
                 PUSH_I32(*(uint32 *)global_addr);
 #else
-                if (!wasm_is_type_reftype(global->type))
+                if (!wasm_is_type_reftype(global->type)) {
                     PUSH_I32(*(uint32 *)global_addr);
-                else
+                }
+                else if (wasm_is_reftype_i31ref(global->type)) {
+                    PUSH_I31REF(GET_REF_FROM_ADDR((uint32 *)global_addr));
+                }
+                else {
                     PUSH_REF(GET_REF_FROM_ADDR((uint32 *)global_addr));
+                }
```

Comment:
Should also handle GET_LOCAL:
```C
                       if (wasm_is_type_reftype(local_type)) {
                            PUSH_REF(
                                GET_REF_FROM_ADDR(frame_lp + local_offset));
                        }
```
And SET_LOCAL:
```C
                       if (wasm_is_type_reftype(local_type)) {
                            PUT_REF_TO_ADDR(frame_lp + local_offset, POP_REF());
                        }
```
No need to set flag for stack/local for i31ref

Reply:
Seems GET_LOCAL already processed? L2657~L2666
And SET_LOCAL is POP_REF, don't need special process

Reply:
OK